controllers/etcdpeer_controller.go

```diff
@@ -270,6 +291,82 @@ func (r *EtcdPeerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, nil
 	}
 
+	if peer.ObjectMeta.DeletionTimestamp.IsZero() {
+		// Add the PVC cleanup finalizer when we first see the peer
+		if !hasPvcDeletionFinalizer(peer) {
```

Comment:
I think we should avoid adding the finalizer unless it is absolutely required, i.e. at the time of a deletion beginning. We use finalizers on Challenge resources in cert-manager and people run into issues where they've stopped our controller (i.e. by deleting the namespace) and they have consequently blocked deletion of any Challenge resources across the cluster, as nothing is there to 'finalize' them.